api/provision/start-provision-job.ts

```diff
@@ -0,0 +1,101 @@
+import { sfnClient } from "../../utils/aws-sdk/step-functions";
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, ErrorStatusCode, OtherErrorResponse, SuccessStatusCode } from "../../utils/response";
+import {
+  CspInvocation,
+  ProvisionRequest,
+  provisionRequestSchema,
+  ProvisionRequestType,
+} from "../../models/provisioning-jobs";
+import { wrapSchema } from "../../utils/middleware/schema-wrapper";
+import { errorHandlingMiddleware } from "../../utils/middleware/error-handling-middleware";
+import middy from "@middy/core";
+import validator from "@middy/validator";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import xssSanitizer from "../../utils/middleware/xss-sanitizer";
+import { IpCheckerMiddleware } from "../../utils/middleware/ip-logging";
+import { CloudServiceProvider } from "../../models/cloud-service-providers";
+import { HttpMethod } from "../../lib/http";
+import { CORS_CONFIGURATION, MIDDY_CORS_CONFIGURATION } from "../../utils/cors-config";
+import { REQUEST_BODY_INVALID } from "../../utils/errors";
+import { cspPortfolioIdChecker } from "../../utils/middleware/check-csp-portfolio-id";
+
+const SFN_ARN = process.env.SFN_ARN ?? "";
+
+/**
+ * Starts provisioning request from SNOW and starts the Step Function execution
+ *
+ * @param event - POST request from API Gateway with provisioning job properties
+ */
+export async function baseHandler(event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> {
+  try {
+    const cspInvocationJob = transformProvisionJob(event.body as any);
```

Comment:
Fair. With using `middy` which wraps the `baseHandler` and uses that event as types for the different middlewares being used (from my understanding at least). I came up with the below solution/type, for the time being rather than using `any` in the middlewares. 

```typescript 
export interface ILambdaEvent {
  body: ProvisionRequest;
  requestContext: { identity: { sourceIp: string } };
}
```

Reply:
Cool, thanks for the suggestion. I'll make a note of this.